### PR TITLE
Inversion of control

### PR DESCRIPTION
**Pull request by yarinvak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,17 @@
...
     "graphql-yoga": "^1.16.1",
-    "typescript": "^3.0.1"
+    "typescript": "^3.0.1",
+    "glob": "^7.1.2",
+    "graphql": "^0.13.2",
+    "graphql-voyager": "^1.0.0-rc.15",
+    "inversify": "^4.13.0",
+    "inversify-binding-decorators": "^4.0.0",
+    "isomorphic-fetch": "latest",
+    "react": "^16.2.0",
```
Comment (osher-sade) [inline package.json]: i don't think that we need those dependencies

File: schema/utils/schema.creator.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import "reflect-metadata";
+import {provide} from "inversify-binding-decorators";
+import {Container, inject, multiInject} from "inversify";
+import {buildProviderModule} from "inversify-binding-decorators";
+import {merge} from 'lodash';
+import {InjectableResolver, InjectableType} from "../../common/injectableInterfaces";
+import {IResolvers} from "graphql-yoga/dist/types";
+
+export interface ISchemaCreator {
+    generateSchema(): { def: string[], resolvers: IResolvers };
```
Comment (guy120494) [inline schema/utils/schema.creator.ts]: Change def to a better type

File: server/graphql.server.ts [hunk 3]
```diff
@@ -14,21 +12,28 @@ export class PolarisGraphQLServer {
...
         return {
             "origin": "*",
             "methods": "GET,HEAD,PUT,PATCH,POST,DELETE",
             "preflightContinue": false,
             "optionsSuccessStatus": 204
         };
     }
-};
+}
+;
```
Comment (guy120494) [inline server/graphql.server.ts]: Why new line?

File: schema/common/entities/commonEntityInterface.ts [hunk 4]
```diff
@@ -3,7 +3,8 @@ import {InjectableType} from "../../../common/injectableInterfaces";
 import {provide} from "inversify-binding-decorators";
 
 @provide("InjectableType")
-export default class CommonEntityInterface implements InjectableType {
+// @ts-ignore
```
Comment (guy120494) [inline schema/common/entities/commonEntityInterface.ts]: Why?

File: schema/utils/schema.creator.ts [hunk 5]
```diff
@@ -5,24 +5,29 @@ import {Container, inject, multiInject} from "inversify";
...
+import {IResolvers} from "graphql-yoga/dist/types"
+import {ITypeDefinitions} from 'graphql-tools';
 
 export interface ISchemaCreator {
-    generateSchema(): { def: string[], resolvers: IResolvers };
+    generateSchema(): { def: ITypeDefinitions, resolvers: IResolvers };
 }
 
 @provide("ISchemaCreator")
+// @ts-ignore
```
Comment (guy120494) [inline schema/utils/schema.creator.ts]: Y?

File: schema/utils/schema.creator.ts [hunk 6]
```diff
@@ -5,24 +5,29 @@ import {Container, inject, multiInject} from "inversify";
...
 export class SchemaCreator implements ISchemaCreator {
     private types: InjectableType[];
     private resolvers: InjectableResolver[];
 
-    public constructor(@multiInject("InjectableType") types: InjectableType[],
-                       @multiInject("InjectableResolver") resolvers: InjectableResolver[]) {
+    public constructor(
+        // @ts-ignore
```
Comment (guy120494) [inline schema/utils/schema.creator.ts]: Y?

**Pull request merged**